src/controllers/posts_crud_controller.rs

```diff
@@ -155,9 +153,10 @@ pub async fn home_gui() -> impl IntoResponse {
 }
 
 pub async fn create_catgories_form(Form(create_category): Form<CreateCategory>) -> Redirect {
-    let pool = get_connection_for_crud().await?;
+    let pool = get_connection_for_crud().await;
     let m = get_max_id_of_category().await;
-    let category_id = m[0].max + 1;
+    let m2 = get_max(m);
+    let category_id = m2 + 1;
```

Comment:
Seems `m2` is referenced only at line 159 ? 

```suggestion
// let m2 = get_max(m);
let category_id = get_max(m)+1;
```